cpp/src/arrow/ipc/metadata-internal.cc

```diff
@@ -266,7 +266,8 @@ Status MessageBuilder::SetRecordBatch(int32_t length, int64_t body_length,
     const std::vector<flatbuf::FieldNode>& nodes,
     const std::vector<flatbuf::Buffer>& buffers) {
   header_type_ = flatbuf::MessageHeader_RecordBatch;
-  header_ = flatbuf::CreateRecordBatch(fbb_, length, fbb_.CreateVectorOfStructs(nodes),
+  header_ = flatbuf::CreateRecordBatch(fbb_, length, flatbuf::Endianness_Little,
```

Comment:
do you think it is worth adding a  check to make sure the code is running on little endian hardware and return an error status if it isn't?